Fix

StuBz211 opened the pull request.

MyLord916 committed:
поборов асинхронность мы уперлись в таймаут

MyLord916 committed:
Из aiohttp в httpx и обратно
Перебрал структуру модулей, подчистил коментарии и импорты

StuBz211 commented on file_managment/ya_file_manager.py:
```diff
@@ -1,62 +1,48 @@
 
 
 if __name__ == "__main__":
-    asyncio.run(main())
+    pass
```
Ненужно.

MyLord916 replied:
Уьерем

StuBz211 commented on handlers/user_handlers.py:
```diff
@@ -5,18 +5,18 @@
     """Список кнопок на которые должен реагировать хендлер смены директории"""
-    but = [*system_buttons.values(), *get_directories_in(path_list)]
+    but = list(system_buttons.values()) + await YaDisk.get_directories_in(path_list)
+    print(but)
```
ещё нужно?

MyLord916 replied:
Нужно оформить иначе, но вроде бы да, реакция хендлера

StuBz211 commented on file_managment/ya_file_manager.py:
```diff
@@ -1,62 +1,48 @@
+        """Функция составляет корректную запись пути к запрашиваемой директории"""
+        return '/'.join(path) + '/'
+
+    @classmethod
```
а почему решил сделать classmethod , чтоб вызывать класс без иницииализации?

MyLord916 replied:
Не совсем тебя понял, выше статический метод, н же декоратор для метода класса.

StuBz211 replied:
по сути ты перенёс все методы под класс, и используешь их без ининциализации класса.

MyLord916 replied:
Не помню для чего она нужна?

StuBz211 commented on handlers/user_handlers.py:
```diff
@@ -64,7 +62,7 @@ async def download_photo(message: types.Message, bot: bot):
     await bot.download(message.photo[-1], destination=buffer)
 
-    upload_img(io.BytesIO(buffer.read()), set_correct_path(path_list) + file_name)
+    await YaDisk.upload_img(io.BytesIO(buffer.read()), YaDisk.set_correct_path(path_list) + file_name)
```
YaDisk.set_correct_path(path_list) - можно убрать внутрь класса

MyLord916 committed:
Исправил замечания
Убрал точку входа вya_file_manager.py
упаковал переменную со списком для хендлера клавиатуры в сам хендлер
запрятал set_correct_path в классе

MyLord916 committed:
add infinite current gen

MyLord916 committed:
Перенес фильтр директорий для хендлера в utils

MyLord916 committed:
recreated requirements.txt
